### ISSUE TITLE
[Bug]: <TabList> crashes if a <Tab> child is missing

### ISSUE THREAD

**Issue by s100** [carbon-design-system/carbon#15278]
### Package

@carbon/react

### Browser

Edge

### Package version

v1.42.1

### React version

v16.14.0

### Description

We are using `<TabList>` with two `<Tabs>`, except that the first of these tabs only renders conditionally. When we render with only the second tab, the application crashes at this line:

https://github.com/carbon-design-system/carbon/blob/17288ffe6b46c64048ae935bc4ca6757aede4658/packages/react/src/components/Tabs/Tabs.tsx#L454

because `selectedItem` is `0`, but `tabs.current[0]` doesn't exist, only `tabs.current[1]` does.

### Reproduction/example

https://stackblitz.com/edit/github-hmphfb?file=src%2FApp.jsx

### Steps to reproduce

Just load up the example and check your browser console for the error:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'disabled')
    at Object.current (@carbon_react.js?v=ba4d934c:88473:37)
    at @carbon_react.js?v=ba4d934c:88131:21
    at commitHookEffectListMount (chunk-KXIZOLV7.js?v=530226d4:16902:34)
    at commitPassiveMountOnFiber (chunk-KXIZOLV7.js?v=530226d4:18150:19)
    at commitPassiveMountEffects_complete (chunk-KXIZOLV7.js?v=530226d4:18123:17)
    at commitPassiveMountEffects_begin (chunk-KXIZOLV7.js?v=530226d4:18113:15)
    at commitPassiveMountEffects (chunk-KXIZOLV7.js?v=530226d4:18103:11)
    at flushPassiveEffectsImpl (chunk-KXIZOLV7.js?v=530226d4:19484:11)
    at flushPassiveEffects (chunk-KXIZOLV7.js?v=530226d4:19441:22)
    at commitRootImpl (chunk-KXIZOLV7.js?v=530226d4:19410:13)
```

Set `showFirstTab` to `true` to see the two tabs.

Note that if you set `showFirstTab` back to `false`, you *won't* see the error until you refresh - this is because the error is occurring inside of a `useEffectOnce` Hook.

I'm guessing that this would also occur if I set `<Tabs defaultSelectedIndex={1}>` and the second tab was omitted, too.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Application/PAL

IBM App Connect Enterprise (don't know what a PAL is)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by s100** [comment]
@tay1orjones Thank you for the prompt fix.

It looks like the Carbon project tends to close issues when the *source code* is fixed. I think it would make sense to consider modifying the procedure you follow, so that issues don't get closed until the fix has been *published* in a new version of the package in question. As for now I guess I just need to wait and check periodically.

**Comment by s100** [comment]
Seems to be fixed as of `@carbon/react@1.46.1`.
